query/stdlib/influxdata/influxdb/storage.go

```diff
@@ -65,138 +63,27 @@ func (l StaticLookup) Watch() <-chan struct{} {
 	return nil
 }
 
-// source performs storage reads
-type source struct {
-	Source
-
-	reader   Reader
-	readSpec ReadSpec
-	window   execute.Window
-	bounds   execute.Bounds
-
-	currentTime execute.Time
-	overflow    bool
-}
-
-func NewSource(id execute.DatasetID, r Reader, readSpec ReadSpec, bounds execute.Bounds, w execute.Window, currentTime execute.Time, alloc *memory.Allocator) execute.Source {
-	src := &source{
-		reader:      r,
-		readSpec:    readSpec,
-		bounds:      bounds,
-		window:      w,
-		currentTime: currentTime,
-	}
-	src.id = id
-	src.alloc = alloc
-	src.runner = src
-	return src
-}
-
-func (s *source) run(ctx context.Context) error {
-	//TODO(nathanielc): Pass through context to actual network I/O.
-	for tables, mark, ok := s.next(ctx); ok; tables, mark, ok = s.next(ctx) {
-		err := s.processTables(ctx, tables, mark)
-		if err != nil {
-			return err
-		}
-	}
-	return nil
-}
-
-func (s *source) next(ctx context.Context) (TableIterator, execute.Time, bool) {
-	if s.overflow {
-		return nil, 0, false
-	}
-
-	start := s.currentTime - execute.Time(s.window.Period)
-	stop := s.currentTime
-	if stop > s.bounds.Stop {
-		return nil, 0, false
-	}
-
-	// Check if we will overflow, if so we are done after this pass
-	every := execute.Time(s.window.Every)
-	if every > 0 {
-		s.overflow = s.currentTime > math.MaxInt64-every
-	} else {
-		s.overflow = s.currentTime < math.MinInt64-every
-	}
-	s.currentTime = s.currentTime + every
-
-	bi, err := s.reader.Read(
-		ctx,
-		s.readSpec,
-		start,
-		stop,
-		s.alloc,
-	)
-	if err != nil {
-		log.Println("E!", err)
-		return nil, 0, false
-	}
-	return bi, stop, true
-}
-
 type GroupMode int
 
 const (
-	// GroupModeDefault specifies the default grouping mode, which is GroupModeAll.
-	GroupModeDefault GroupMode = 0
 	// GroupModeNone merges all series into a single group.
-	GroupModeNone GroupMode = 1 << iota
-	// GroupModeAll produces a separate table for each series.
-	GroupModeAll
+	GroupModeNone GroupMode = iota
```

Comment:
Is this correct?  I would think that `GroupModeNone` wouldn't do any grouping, i.e., leave the output grouped by the tag keys.

Reply:
I was definitely confused by this, but Flux as it stands right now has no concept of GroupModeNode. It can do group by, or group except, that's it. Storage however defines GroupNone to to be that the result is "all one group", and it can make optimizations for that specific case. To remove the ambiguity, I removed GroupModeAll, because there's no way to specify that type of grouping using a `group()` operation. It is just the default from storage.

Reply:
We should plan to drop the `GroupModeNone` constant completely and have a new physical operation that represents this behavior. After all, it is implemented separately in storage too and therefore should be a separate RPC.